lib/cylc/rundb.py

```diff
@@ -554,6 +554,27 @@ def select_task_job(self, keys, cycle, name, submit_num=None):
         except sqlite3.DatabaseError:
             return None
 
+    def select_task_job_run_times(self, callback):
+        """Select run times of succeeded task jobs grouped by task names.
+
+        Invoke callback(row_idx, row) on each row, where each row contains:
+            [name, run_times_str]
+
+        where run_times_str is a string containing comma separated list of
+        integer run times. This method is used to re-populate elapsed run times
+        of each task on restart.
+        """
+        stmt = (
+            r"SELECT" +
+            r" name," +
+            r" GROUP_CONCAT(" +
+            r"     CAST(strftime('%s', time_run_exit) AS NUMERIC) -" +
+            r"     CAST(strftime('%s', time_run) AS NUMERIC))" +
```

Comment:
I think I'd be inclined to do this calculation in the python rather than an sql statement as sql is generally inefficient about doing these kind of manipulations vs. python. However, probably not going to notice it here.